posthog-react-native/src/storage.ts

```diff
@@ -17,26 +17,35 @@ export class SemiAsyncStorage {
   }
 
   preloadAsync(): Promise<void> {
+    if (this.isPreloaded) {
+      return Promise.resolve()
+    }
+
     if (this._preloadSemiAsyncStoragePromise) {
       return this._preloadSemiAsyncStoragePromise
     }
+
     this._preloadSemiAsyncStoragePromise = this._asyncStorage.getItem(POSTHOG_STORAGE_KEY).then((res) => {
       try {
         const data = res ? JSON.parse(res).content : {}
 
         for (const key in data) {
           this._memoryCache[key] = data[key]
         }
-
-        this.isPreloaded = true
       } catch (e) {
         console.warn(
           "PostHog failed to load persisted data from storage. This is likely because the storage format is. We'll reset the storage.",
           e
         )
+      } finally {
+        this.isPreloaded = true
```

Comment:
It's a bit unobvious but basically I realised even if parsing fails we should call it "preloaded" because there is no other state that can be handled other than resetting the state